### PR TITLE
WIP: use the atom-renderer library

### PR DESCRIPTION
**Pull request by tomrf1**
Migrates to using https://github.com/guardian/atom-renderer, which is designed to support atom rendering across platforms.

TODO - platform:
- [ ] Does not currently use `JSMinifier` as it fails to compile
- [ ] Needs correct fonts
- [ ] Support AMP
- [ ] Provide prod version of services
- [ ] Define a placeholder for atom factories
- [ ] Solve the escaping issue

TODO - journalism:
- [ ] Remove old html/css

### CODE REVIEW

File: common/app/model/meta.scala [hunk 1]
```diff
@@ -173,6 +175,18 @@ object MetaData {
...
+      Seq(
+        atoms.guides.map(_ => AtomType.Guide),
+        atoms.qandas.map(_ => AtomType.Qanda),
+        atoms.profiles.map(_ => AtomType.Profile),
+        atoms.timelines.map(_ => AtomType.Timeline)
+      ).flatten
+    }
+
+    val atomsCSS: Seq[String] = ArticleAtomRenderer.getCSS(atomTypes)
+    val atomsJS: Seq[String] = ArticleAtomRenderer.getJS(atomTypes)
```
Comment (TBonnin) [inline common/app/model/meta.scala]: can those be a `String` directly?

Comment (TBonnin) [inline common/app/model/meta.scala]: I don't think I understand why atom css and js belongs in `metadata`. Can you explain? Thank you

File: common/app/templates/inlineJS/nonBlocking/atomsJs.scala.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@(page: model.Page)(implicit context: model.ApplicationContext)
+
+@page.metadata.atomsJS.mkString
```
Comment (TBonnin) [inline common/app/templates/inlineJS/nonBlocking/atomsJs.scala.js]: why another template with only one line?

File: article/app/pages/StoryHtmlPage.scala [hunk 3]
```diff
@@ -11,19 +11,20 @@ import views.html.fragments._
...
-  def allStyles(implicit applicationContext: ApplicationContext): Styles = new Styles {
+  def allStyles(implicit applicationContext: ApplicationContext, page: Page): Styles = new Styles {
     override def criticalCssLink: Html = criticalStyleLink("content")
     override def criticalCssInline: Html = criticalStyleInline(Html(common.Assets.css.head(None)))
     override def linkCss: Html = stylesheetLink("stylesheets/content.css")
     override def oldIECriticalCss: Html = stylesheetLink("stylesheets/old-ie.head.content.css")
     override def oldIELinkCss: Html = stylesheetLink("stylesheets/old-ie.content.css")
     override def IE9LinkCss: Html = stylesheetLink("stylesheets/ie9.head.content.css")
     override def IE9CriticalCss: Html = stylesheetLink("stylesheets/ie9.content.css")
+    override def atomsCSS: Option[Html] = Some(atomsCSSInline(page))
```
Comment (TBonnin) [inline article/app/pages/StoryHtmlPage.scala]: Unless you need the atomCss to be in a specific place you can keep things simple by not adding it here but after the `style(allStyles)` in the `html` [method](https://github.com/guardian/frontend/pull/17968/files#diff-814f684a6185ffdf5d7bd0d8e414604eR46):
```
htmlTag(
       headTag(
         titleTag(),
         metaData(),
         head,
         styles(allStyles),
         atomsCSSInline(),
         fixIEReferenceErrors(),
```

File: common/app/templates/inlineJS/nonBlocking/atomsJs.scala.js (hunk 2 above)
Comment (tomrf1) [reply common/app/templates/inlineJS/nonBlocking/atomsJs.scala.js]: sorry, this file is very WIP!

File: common/app/model/meta.scala (hunk 1 above)
Comment (regiskuckaertz) [reply common/app/model/meta.scala]: Sorry that's me, I did this to make it work even though I knew there would be a better solution eventually. That is where you come in 😅

File: common/app/templates/inlineJS/nonBlocking/atomsJs.scala.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+@import play.twirl.api.HtmlFormat
+@(page: model.Page)(implicit context: model.ApplicationContext)
+(function () {
+  function loadAtoms() {
+    @* 
+     * each atomsJS element looks like this:
+     *   this.profile=(function(...){...}())
+     * we can just bind this to an arbitrary object that we can pass around.   
+     *@
+    @page.metadata.atomsJS.mkString(" ")
```
Comment (regiskuckaertz) [inline common/app/templates/inlineJS/nonBlocking/atomsJs.scala.js]: For some reason, the javascript is escaped (quotes are backslashed), but even wrapping in `HtmlFormat.raw` does not change anything 🤔 😲

File: common/app/model/meta.scala (hunk 1 above)
Comment (TBonnin) [reply common/app/model/meta.scala]: no pb. Ping me when it is ready for review.

**Comment by regiskuckaertz** [comment]
@TBonnin @sndrs I think all the building blocks are there, now it needs your magic touch ⚡️ 

Can we meet next week so we can walk you through it and then you can hopefully plan the work on the native interface?

**Comment by regiskuckaertz** [comment]
Closing this one because easier to start from scratch with the npm package stuff

**Pull request closed without merging**